readthedocs/ssh/models.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import division, print_function, unicode_literals
+
+import base64
+import hashlib
+import json
+
+from builtins import zip
+from django.db import models
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
+
+from readthedocs.projects.models import Project
+
+from .mixins import SSHKeyGenMixin
+from .querysets import SSHKeyQuerySet
+
+
+@python_2_unicode_compatible
+class SSHKey(SSHKeyGenMixin, models.Model):
+
+    pub_date = models.DateTimeField(_('Publication date'), auto_now_add=True)
+
+    public_key = models.TextField(
+        _('Public SSH Key'),
+        help_text='Add this to your version control to give us access.',
+    )
+    private_key = models.TextField(_('Private SSH Key'))
```

Comment:
How do we make this an encrypted field? That is, this should definitely be encrypted somehow, but what attacks do we need to prevent against. Encrypting fields with a per-instance key protects a database leak, but doesn't help leaking through application bugs. Can we do per-user/per-org encryption? Any prior art for managing this?

Reply:
Yeah, i'm not really sure how to address this. Saving a private key to the database won't net us anything as an attack would just be able to get the private key anyways.

Maybe we can talk more about this, I feel this is a pretty complex problem. I tried looking into how GitLab might do this, but I'm not familiar enough with GitLab or ruby applications to understand what they do (if anything) with private keys.